dev/AppLifecycle/AppInstance.cpp

```diff
@@ -308,51 +316,67 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
 
             // All specific launch types are encoded as a URI and transported as a
             // protocol, except the catch-all LaunchActivatedEventArgs case.
-            if (!contractId.empty() && (CompareStringOrdinal(contractId.c_str(), -1, c_protocolArgumentString, -1, TRUE) == CSTR_EQUAL))
+            if (!contractArgument.empty())
             {
-                kind = ExtendedActivationKind::Protocol;
-                auto args = make<ProtocolActivatedEventArgs>(contractData);
-                data = args.as<IInspectable>();
-
-                // Encoded launch is a protocol launch where the argument data is
-                // encapsulated in the Uri Query data.  We handle that here and
-                // try to return the correct IActivatedEventArgs type that is
-                // encoded in the data rather than the IProtocolActivatedEventArgs
-                // itself.
-                auto protocolArgs = args.as<IProtocolActivatedEventArgs>();
-                if (IsEncodedLaunch(protocolArgs.Uri()))
+                if (contractData.empty())
                 {
-                    std::tie(kind, data) = GetEncodedLaunchActivatedEventArgs(protocolArgs);
+                    // If the contractData is empty, handle any aliased encoded launches.
+                    if (CompareStringOrdinal(contractArgument.data(), static_cast<int>(contractArgument.size()), L"ReunionPushServer", -1, TRUE) == CSTR_EQUAL)
+                    {
+                        contractData = GenerateEncodedLaunchUri(L"App", c_pushContractId);
+                        contractArgument = c_protocolArgumentString;
+                    }
                 }
-            }
 
-            // Haven't set the kind yet, and so let's default to Launch.
-            if (kind == ExtendedActivationKind::Launch)
-            {
-                data = make<LaunchActivatedEventArgs>(commandLine).as<IInspectable>();
+                if (CompareStringOrdinal(contractArgument.c_str(), -1, c_protocolArgumentString, -1, TRUE) == CSTR_EQUAL)
+                {
+                    kind = ExtendedActivationKind::Protocol;
+                    auto args = make<ProtocolActivatedEventArgs>(contractData);
+                    data = args;
+
+                    // Encoded launch is a protocol launch where the argument data is
+                    // encapsulated in the Uri Query data.  We handle that here and
+                    // try to return the correct IActivatedEventArgs type that is
+                    // encoded in the data rather than the IProtocolActivatedEventArgs
+                    // itself.
+                    if (IsEncodedLaunch(args.Uri()))
+                    {
+                        std::tie(kind, data) = GetEncodedLaunchActivatedEventArgs(args);
+                    }
+
+                    foundArgs = true;
+                }
             }
         }
 
+        // All scenarios should just be marked as Launch.
+        if (!foundArgs)
```

Comment:
>foundArgs [](http://example.com/codeflow?start=13&length=9)

Looks like this variable is superfluous. `foundArgs` is true exactly when `data` is non-null.